pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -45,9 +45,13 @@ const (
 	RequestQueueAdminPortName string = "queueadm-port"
 
 	// RequestQueueMetricsPortName specifies the port name to use for metrics
-	// emitted by queue-proxy.
+	// emitted by queue-proxy for autoscaler.
 	RequestQueueMetricsPortName = "queue-metrics"
 
+	// PublicQueueMetricsPortName specifies the port name to use for metrics
+	// emitted by queue-proxy for end user.
+	PublicQueueMetricsPortName = "queue-metrics-p"
```

Comment:
nit: change "queue-metrics-p" to "public-queue-metrics"?

Reply:
or may be "user-metrics"?

Reply:
public-queue-metrics was the first thing I did and it failed because k8s have a limit of 15 chars on port name :) Will go with user-metrics.